gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -264,6 +273,24 @@ static boolean isOnComputeEngine() {
     return false;
   }
 
+  @VisibleForTesting
+  ChannelCredentials createMtlsChannelCredentials() throws IOException {
+    if (mtlsProvider.useMtlsClientCertificate()) {
+      try {
+        KeyStore mtlsKeyStore = mtlsProvider.getKeyStore();
+        if (mtlsKeyStore != null) {
+          KeyManagerFactory factory =
+              KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+          factory.init(mtlsKeyStore, new char[] {});
+          return TlsChannelCredentials.newBuilder().keyManager(factory.getKeyManagers()).build();
+        }
+      } catch (GeneralSecurityException e) {
```

Comment:
why wrap this here? Should be possible to catch and transform higher up if (and only if) absolutely necessary. It's better to keep exceptions more semantically accurate as long as possible.

Reply:
This function and all the upstream functions only throw IOException, so it is impossible not to wrap it.

Reply:
Yes, but you could wrap or handle it there instead of here. 

Another way of thinking about it: if the upstream methods aren't declared to handle a security exception, they're saying they didn't plan for calling any methods that could have security issues.

Reply:
Done. Wrapped the exception in the caller instead of here.